### ISSUE TITLE
Indexer can't handle pages with sys_language_uid = -1

### ISSUE THREAD

**Issue by sturm-solutions** [teaminmedias-pluswerk/ke_search#197]
Hey everyone,
first things first: Great work! Great extension!

Issue:
Pages with sys_language_uid = -1 will not be indexed. The problem exists in method checkIfpageShouldBeIndexed and addLocalizedPagesToCache in file Classes/indexer/types/class.tx_kesearch_indexer_types_page.php:

```php
public function checkIfpageShouldBeIndexed($uid, $language_uid) { // $uid = 123, $language_uid = -1
	$index = true;

	// ...

	if (!in_array($this->cachedPageRecords[$language_uid][$uid]['doktype'], $this->indexDokTypes)) {
		// $this->cachedPageRecords[-1][123]['doktype'] is NULL everytime, because it doesn't exist 
		$index = false; // everytime false, because in_array fails with cache $this->cachedPageRecords[-1]
	}
	return $index;
}
```
So pages with sys_language_uid = -1 will not be added in method addLocalizedPagesToCache, because foreach statement only handles existing sys languages:

```php
public function addLocalizedPagesToCache($pageRow) {
	// ...

	// create entry in cachedPageRecods for additional languages, skip default language 0
	foreach ($this->sysLanguages as $sysLang) { // doesn't work with sys_language_uid = -1
		if ($sysLang[1] > 0) { // doesn't work with sys_language_uid = -1
				list($pageOverlay) = \TYPO3\CMS\Backend\Utility\BackendUtility::getRecordsByField('pages_language_overlay', 'pid', $pageRow['uid'], 'AND sys_language_uid=' .intval($sysLang[1]));
			if ($pageOverlay) {
				$this->cachedPageRecords[$sysLang[1]][$pageRow['uid']] = \TYPO3\CMS\Core\Utility\GeneralUtility::array_merge($pageRow, $pageOverlay);
			}
		}
	}
}
```

Can you tell me if you plan to fix this mentioned issue in the future?

Best regards and thanks

**Comment by sturm-solutions** [comment]
I will fork your repo and take a look at it. When finished I'll provide a pull request. Can't say when it will be finished, but try to provide it asap.

**Comment by sturm-solutions** [comment]
Fixed and pull request provided - enjoy!

**Comment by christianbltr** [comment]
I'm not sure if this is really an issue. As far as I can see it is not possible to create a page with sys_language_uid = - 1. Editors can only create pages in the default language (sys_language_uid = 0) or translations of that page in given languages (sys_language_uid > 0).

Only content elements can have sys_language_uid = -1 and the show up in all language versions of a page. Indexing of those content elements has been fixed in #198

**Comment by christianbltr** [comment]
As commented before this issue should be fixed.
